src/category/category.service.test.ts

```diff
@@ -0,0 +1,214 @@
+import { CategoryService } from './category.service';
+import { PrismaService } from '../database/prisma.service';
+import { NotFoundException } from '@nestjs/common';
+import { Prisma, Category } from '@prisma/client';
+import { CreateCategoryDto } from './dto/create-category.dto';
+import { UpdateCategoryDto } from './dto/update-category.dto';
+import { PrismaError } from '../database/prisma-error.enum';
+import { CategoryDto } from './dto/category.dto';
+
+describe('The CategoryService', () => {
+  let categoryService: CategoryService;
+  let findManyCategoryMock: jest.Mock;
+  let findUniqueCategoryMock: jest.Mock;
+  let createCategoryMock: jest.Mock;
+  let updateCategoryMock: jest.Mock;
+  let deleteCategoryMock: jest.Mock;
+  let categoriesArray: Category[];
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    findManyCategoryMock = jest.fn();
+    findUniqueCategoryMock = jest.fn();
+    createCategoryMock = jest.fn();
+    updateCategoryMock = jest.fn();
+    deleteCategoryMock = jest.fn();
+    const module = await Test.createTestingModule({
+      providers: [
+        CategoryService,
+        {
+          provide: PrismaService,
+          useValue: {
+            category: {
+              findMany: findManyCategoryMock,
+              findUnique: findUniqueCategoryMock,
+              create: createCategoryMock,
+              update: updateCategoryMock,
+              delete: deleteCategoryMock,
+            },
+            amenity: {
+              findUnique: findUniqueCategoryMock,
+            },
+          },
+        },
+      ],
+    }).compile();
+
+    categoryService = module.get(CategoryService);
+    categoriesArray = [
+      { id: 1, name: 'Amenities' },
+      { id: 2, name: 'Handicap Accessibility' },
+    ];
+  });
+
+  describe('when getAll is called', () => {
+    describe('and categories exist', () => {
+      beforeEach(() => {
+        findManyCategoryMock.mockResolvedValue(categoriesArray);
+      });
+      it('should return the categories', async () => {
+        const result = await categoryService.getAll();
+        expect(result).toEqual(categoriesArray);
+      });
+    });
+    describe('and no categories exist', () => {
+      beforeEach(() => {
+        findManyCategoryMock.mockResolvedValue([]);
+      });
+      it('should return an empty array', async () => {
+        const result = await categoryService.getAll();
+        expect(result).toEqual([]);
+      });
+    });
+  });
+
+  describe('when getOne is called', () => {
+    describe('and the category exists', () => {
+      beforeEach(() => {
+        findUniqueCategoryMock.mockResolvedValue(categoriesArray[0]);
+      });
+      it('should return the category', async () => {
+        const result = await categoryService.getOne(categoriesArray[0].id);
+        expect(result).toEqual(categoriesArray[0]);
+      });
+    });
+    describe('and the category does not exist', () => {
+      beforeEach(() => {
```

Comment:
there is no point of wrapping it in beforeEach, there is just a single test here, so you can put the mocking just above it statement